test_util/installer_api_test.py

```diff
@@ -3,58 +3,77 @@
 import abc
 import json
 import os
-import subprocess
+from subprocess import CalledProcessError
 
 import requests
 import yaml
 from retrying import retry
 
+from ssh.ssh_tunnel import SSHTunnel, run_scp_cmd, run_ssh_cmd
+
 
 class AbstractDcosInstaller(metaclass=abc.ABCMeta):
 
     def __init__(self):
         self.offline_mode = False
 
-    def setup_remote(self, host, ssh_user, ssh_key_path, installer_path, download_url):
-        # Refresh ssh info
-        self.url = "http://{}:9000".format(host)
+    def setup_remote(
+            self, tunnel, installer_path, download_url,
+            host=None, ssh_user=None, ssh_key_path=None):
+        """Creates a light, system-based ssh handler
+        Args:
+            tunnel: SSHTunnel instance to avoid recreating SSH connections.
+                If set to None, ssh_user, host, and ssh_key_path must be
+                set and one-off connections will be made
+            installer_path: (str) path on host to download installer to
+            download_url: (str) URL that installer can be pulled from
+            host: (str) where the installer will be downloaded to
+            ssh_user: (str) user with access to host
+            ssh_key_path: (str) path to valid ssh key for ssh_user@host
+        """
         self.installer_path = installer_path
-        self.host = host
-        ssh_opts = [
-                '-i', ssh_key_path,
-                '-oConnectTimeout=10',
-                '-oStrictHostKeyChecking=no',
-                '-oUserKnownHostsFile=/dev/null',
-                '-oBatchMode=yes',
-                '-oPasswordAuthentication=no']
-
-        def ssh(cmd):
-            assert isinstance(cmd, list)
-            return ['/usr/bin/ssh']+ssh_opts+['{}@{}'.format(ssh_user, host)]+cmd
-
-        def scp(src, dst):
-            return ['/usr/bin/scp']+ssh_opts+[src, '{}@{}:{}'.format(ssh_user, host, dst)]
+        if not tunnel:
```

Comment:
Nit: can we have `if tunnel; else` and flip the blocks? The negative logic is ever so slightly more complicated to read, but it's in no way a blocker.